### DISCUSSION TITLE
When there would be a version 1.0?

### DISCUSSION THREAD

**Discussion by Gr3Kidd3r** [fastapi/fastapi#8651] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
There's no code to be included.
```

### Description

I read in the docs that at the moment FastAPI is production ready and is used by big tech companies, but I'm asking when there will be a release for version 1.0? And if there is one, what is to be expected from that version as I think FastAPI is using semantic versioning.

### Operating System

Linux

### Operating System Details

Ubuntu 20.04 LTS

### FastAPI Version

0.70.0

### Python Version

3.7.11

**Comment by Kludex** [comment, marked as answer]
Based on my talks with @tiangolo, this is the checklist:
- API Reference (and some internal cleaning for it)
- Starlette stable release (1.0)
- Uvicorn stable release (1.0)

It's likely that I forgot about some other box, or that I'm not aware. But the core of our conversation is that the "underneath packages should be stable".

As for the Starlette 1.0 release, we are missing:
- HTTP/2 support
- HTTPX based TestClient

You can check more about it on https://github.com/encode/starlette/issues/51.

As for Uvicorn 1.0 release, we are missing more things. Check more about it on https://github.com/encode/uvicorn/milestone/1.

PRs are more than welcome on Starlette and Uvicorn. :)

NOTE: This is not an official statement.

**Reply by jpmckinney** [reply]
Looks like Starlette is almost ready for 1.0. There's a draft PR: https://github.com/encode/starlette/pull/1888

Uvicorn seems to now only be missing 100% coverage: https://github.com/encode/uvicorn/issues/102 Looking at a recent GitHub Actions runs on the `master` branch, coverage is at 98.86% with 60 lines missed.

FastAPI seems to still lack an API Reference.

**Reply by Kludex** [reply]
Also waiting for pydantic v2

**Reply by jpmckinney** [reply]
For Pydantic, can track here: https://github.com/pydantic/pydantic/issues/4887 Alpha releases of v2 are out.

**Comment by Gr3Kidd3r** [comment]
@Kludex Thanks for your response.

**Comment by ghandic** [comment]
How much of the openapi specification would we also need to tick off? These could have breaking changes (thinking deepObject query encoding) and a few other extra features
